InterviewPrep/chapter01/question0101.cs

```diff
@@ -1,26 +1,91 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
-
-/**
-   Implement an algorithm to determine if a string has all unique characters.
-**/
-namespace InterviewPrep.chapter01
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright company="" file="Question0101.cs">
+//   
+// </copyright>
+// <summary>
+//   Question 0101: Does the string have unique characters?
+// </summary>
+// 
+// --------------------------------------------------------------------------------------------------------------------
+namespace InterviewPrep.Chapter01
 {
-    class question0101
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading.Tasks;
+
+    /// <summary> Question 0101: Does the string have unique characters? </summary>
+    public class Question0101
     {
+        /// <summary> Does the string have unique characters.    </summary>
+        /// <param name="s"> The s.   </param>
+        /// <exception cref="AggregateException"> The exception that contains all the individual exceptions thrown on all threads.  </exception>
+        /// <exception cref="OverflowException"> The dictionary already contains the maximum number of elements,     <see cref="F:System.Int32.MaxValue"/>. </exception> 
+        /// <exception cref="ArgumentNullException"> <paramref name="key"/> is a null reference (Nothing in Visual Basic).  </exception>
+        /// <returns> The <see cref="bool"/>.  </returns>
+        public static bool HasUniqueCharacters(string s)
+        {
+            // Contract.Requires(s != null);
+            if (CornerCase(s))
+            {
+                return true;
+            }
+
+            var cd = new ConcurrentDictionary<char, bool>();
+            var unique = true;
+            Parallel.ForEach(s, c =>
+                    {
+                        if (cd.ContainsKey(c))
+                        {
```

Comment:
One comment is on using multithreading here - multi threading is not really useful for CPU intensive problems. The cost of spinning a thread in this case might be too expensive. 

In general the code looks good.